build-scripts/k8s_release.py

```diff
@@ -18,72 +18,131 @@
 
 
 def _url_get(url: str) -> str:
-    r = requests.get(url, timeout=5)
-    r.raise_for_status()
-    return r.text
+    """Make a GET request to the given URL and return the response text."""
+    response = requests.get(url, timeout=5)
+    response.raise_for_status()
+    return response.text
+
+
+def is_stable_release(release: str) -> bool:
+    """Check if a Kubernetes release tag is stable (no pre-release suffix).
+
+    Args:
+        release: A Kubernetes release tag (e.g. 'v1.30.1', 'v1.30.0-alpha.1').
+
+    Returns:
+        True if the release is stable, False otherwise.
+    """
+    return "-" not in release
 
 
 def get_k8s_tags() -> List[str]:
-    """Retrieve semantically ordered k8s releases, newest to oldest."""
+    """Retrieve semantically ordered Kubernetes release tags from GitHub.
+
+    Returns:
+        A list of release tag strings sorted from newest to oldest.
+
+    Raises:
+        ValueError: If no tags are retrieved.
+    """
     response = _url_get(K8S_TAGS_URL)
     tags_json = json.loads(response)
-    if len(tags_json) == 0:
+    if not tags_json:
         raise ValueError("No k8s tags retrieved.")
     tag_names = [tag["name"] for tag in tags_json]
-    # Github already sorts the tags semantically but let's not rely on that.
     tag_names.sort(key=lambda x: Version(x), reverse=True)
     return tag_names
 
 
-# k8s release naming:
-# * alpha:  v{major}.{minor}.{patch}-alpha.{version}
-# * beta:   v{major}.{minor}.{patch}-beta.{version}
-# * rc:     v{major}.{minor}.{patch}-rc.{version}
-# * stable: v{major}.{minor}.{patch}
-def is_stable_release(release: str):
-    return "-" not in release
+def get_latest_stable() -> str:
+    """Get the latest stable Kubernetes release tag.
 
+    Returns:
+        The latest stable release tag string (e.g., 'v1.30.1').
 
-def get_latest_stable() -> str:
-    k8s_tags = get_k8s_tags()
-    for tag in k8s_tags:
+    Raises:
+        ValueError: If no stable release is found.
+    """
+    for tag in get_k8s_tags():
         if is_stable_release(tag):
             return tag
-    raise ValueError("Couldn't find stable release, received tags: %s" % k8s_tags)
+    raise ValueError("Couldn't find a stable release.")
 
 
-def get_latest_release() -> str:
-    k8s_tags = get_k8s_tags()
-    return k8s_tags[0]
+def get_latest_releases_by_minor() -> Dict[str, str]:
+    """Map each minor Kubernetes version to its latest release tag.
+
+    Returns:
+        A dictionary mapping minor versions (e.g. '1.30') to the
+        latest (pre-)release tag (e.g. 'v1.30.1').
+    """
+    latest_by_minor: Dict[str, str] = {}
+    version_regex = re.compile(r"^v?(\d+)\.(\d+)\..+")
+
+    for tag in get_k8s_tags():
+        match = version_regex.match(tag)
+        if not match:
+            continue
+        major, minor = match.groups()
+        key = f"{major}.{minor}"
```

Comment:
Nit: I'd rather parse the version using `packaging.version.Version` instead of regex. It's cleaner / easier to read.